### DISCUSSION TITLE
Code server install on CentO7/nginx

### DISCUSSION THREAD

**Discussion by highosilver** [coder/code-server#2553] [Q&A]
I'm trying to get code server on a VPS running CentOS 7 with an nginx reverse proxy configuration. I was able to get the install done relatively easily, however the issue I run into is when visiting my chosen domain I receive an index of page. 

I think this may be due to nginx failing to pass the connection to code server on my chosen port. 

/nginx/code-server.conf:

` server {
    listen 80;
    listen [::]:80;

    server_name code-server.{domain}.com;

    location / {
        proxy_pass http://localhost:4479/;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection upgrade;
        proxy_set_header Accept-Encoding gzip;
    }
} `

And /lib/systemd/system/code-server.service:

` [Unit]
Description=code-server
After=nginx.service

[Service]
Type=simple
Environment=PASSWORD=}?$C5he(!]p_#aC%
ExecStart=/usr/bin/code-server --bind-addr 127.0.0.1:4479 --user-data-dir /var/lib/code-server --auth password
Restart=always

[Install]
WantedBy=multi-user.target `

On my system I have nginx listening on ports 80 and 443, with apache listening on 8080 and 8443. 

I've tried running
` setsebool httpd_can_network_connect 1 -P `
However it appears my host has SELinux disabled at the node level.

If I set my bind address to 0.0.0.0:Port, I can connect using my dedicated IP and the port as my URL, leading to believe my issue is with nginx passing the connection to the code-server. 

Any help or insight would be greatly appreciated!

**Comment by code-asher** [comment]
Hmm yeah your configuration looks good to me. Do the NGINX error logs
say anything when you try to access the URL?

**Reply by code-asher** [reply]
Also are you including `/nginx/code-server.conf` from the main nginx config or something? If not you might need to move it into `/etc/nginx/sites-available/code-server` and reload NGINX.

**Reply by code-asher** [reply]
Would also need to enable it: `ln -s ../sites-available/code-server /etc/nginx/sites-enabled/code-server`

More info here: https://github.com/cdr/code-server/blob/v3.8.0/doc/guide.md#nginx
